application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,182 @@
+            .queue();
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    private static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX;
+    }
+
+    private RestAction<User> createForumPost(ModalInteractionEvent event, User originalUser) {
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+        ForumChannel questionsForum = getHelperForum(event.getJDA());
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+        ForumTag javaTag = tags.get(0);
+
+        return questionsForum.createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
+            .map(ForumPost::getMessage)
+            .flatMap(message -> message.reply("Original Post from " + originalUser.getAsMention()))
+            .map(sent -> originalUser);
+
+    }
+
+    private RestAction<ForumChannel> dmUser(MessageChannelUnion sourceChannel, User originalUser) {
+
+        return originalUser.openPrivateChannel()
+            .flatMap(channel -> channel.sendMessage(
+                    "Hi, TJ here. You are getting this message " + "because you tried to ask a "
+                            + "question in wrong chat, please read rules - Together Java"))
+            .onErrorFlatMap(error -> sourceChannel
+                .sendMessage("Due to failed dm interaction, you are getting "
+                        + "this message here. Please read rules about asking questions - Together Java"
+                        + originalUser.getAsMention()))
+            .onSuccess(failed -> logger.warn("failed to dm user {} with id: {}",
+                    originalUser.getName(), originalUser.getId()))
+            .map(dm -> getHelperForum(dm.getJDA()));
+
+    }
+
+    private RestAction<Void> deleteOriginalMessage(JDA jda, String channelId, String messageId) {
+        return jda.getTextChannelById(channelId)
+            .deleteMessageById(messageId)
+            .onSuccess(deleted -> logger.warn(
+                    "message with id: {} has been deleted due to transfer action being successful",
+                    messageId));
```

Comment:
no need to log

Reply:
if you mean "logging the deleted message", isn't it a good idea to store a deleted message. It's an important event? no?

Reply:
message deletes are auto-logged in discords own log system. dont worry :)

note for the future: if at all, it wouldnt be a warning, but rather info or debug or trace

Reply:
Idk why i put it to "warn", it's just part of the process🤦‍♂️